wp_api_integration_tests/tests/test_cookie_authentication_immut.rs

```diff
@@ -0,0 +1,70 @@
+use wp_api::{WpApiClient, WpAuthentication, WpLoginCredentials};
+
+use wp_api_integration_tests::*;
+
+use serial_test::serial;
+
+#[tokio::test]
+#[serial]
+async fn test_cookie_authentication() {
+    let login = WpLoginCredentials {
+        username: TestCredentials::instance().admin_username.to_string(),
+        password: TestCredentials::instance()
+            .admin_account_password
+            .to_string(),
+    };
+    let request_executor = std::sync::Arc::new(AsyncWpNetworking::with_cookie_store());
+    let client = WpApiClient::new(
+        test_site_url(),
+        WpAuthentication::UserAccount { login },
+        request_executor,
+    );
+    let user = client
+        .users()
+        .retrieve_me_with_edit_context()
+        .await
+        .assert_response();
+    assert_eq!(user.id, FIRST_USER_ID);
+}
+
+#[tokio::test]
+#[serial]
+async fn test_fail_with_incorrect_password() {
+    let login = WpLoginCredentials {
+        username: TestCredentials::instance().admin_username.to_string(),
+        password: "incorrect".to_string(),
+    };
+    let request_executor = std::sync::Arc::new(AsyncWpNetworking::with_cookie_store());
+    let client = WpApiClient::new(
+        test_site_url(),
+        WpAuthentication::UserAccount { login },
+        request_executor,
+    );
+    client
+        .users()
+        .retrieve_me_with_edit_context()
+        .await
+        .assert_wp_error(wp_api::WpErrorCode::Unauthorized);
+}
+
+#[tokio::test]
+#[serial]
+async fn test_fail_without_cookie_store() {
```

Comment:
This is an interesting test case, because it doesn't actually test the `wp_api` implementation as it rather tests `AsyncWpNetworking`. I don't think it's wrong to have this test, because it showcases that the request executor implementation matters.

There are a couple things to take away from this:
1. Our design is at least a little lacking, because we rely on the specifics of a client side implementation without enforcing it. Ideally, we'd not rely on a client side implementation at all, but if it's unavoidable, we'd at least have a way to enforce it.
2. Until we (can) address `(1)` we need to document it.